### PR TITLE
add hack/arktos_cherrypick.sh

### PR DESCRIPTION
**Pull request by sonyafenge**
PLease add "CHERRY_PICK=true" before make update to enable this

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>

> /kind documentation

**What this PR does / why we need it**:
Add hack/arktos_cherrypick.sh. it will add comments "File modified by cherrypick from Kubernetes on $(date +'%m/%d/%Y')" when PR is cherry-picked from community Kubernetes. 
```
hack/arktos_cherrypick.sh
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
```
Scenarios #1: File already exist in Arktos, modified by cherrypick
Existing:         Copyright 2016 The Kubernetes Authors.
New added:   Copyright 2020 Authors of Arktos - file modified.
Add cherrypick comments by hack/arktos_cherrypick.sh:  
New added:   File modified by cherrypick from kubernetes on 02/18/2021
```
``` 
Scenarios #1-1: File already exist in Arktos, and modified before cherrypick
Existing:                 Copyright 2016 The Kubernetes Authors.
Existing:                 Copyright 2020 Authors of Arktos - file modified.
``` 
```
Scenarios #2: File already exist in Arktos, modified by Arktos feature request
Existing:        Copyright 2016 The Kubernetes Authors.
New added:   Copyright 2020 Authors of Arktos - file modified.
```
```
Scenarios #3: File doesn’t exist in Arktos, new added by cherrypick
Existing:         Copyright 2016 The Kubernetes Authors.
New added:   Copyright 2020 Authors of Arktos - file modified.
Add cherrypick comments by hack/arktos_cherrypick.sh:  
New added: File modified by cherrypick from kubernetes on 02/18/2021
Questions: Is Kubernetes copyrights necessary? This file should put to hack/arktos_copyright_copied_modified_k8s_files.
```
``` 
Scenarios #4: File doesn’t exist in Arktos, new added after auto generate per cherrypick commit
New generated:      Copyright 2020 Authors of Arktos.
Questions: Is Kubernetes copyrights necessary? No legal issue, we can only keep arktos copyright because this auto generated by arktos for now. 
```
**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by Sindica** [comment]
I have a concern to add this as a generic component for make update. Cherry-pick is not regular process we are doing. Also CI tool cannot be updated to have this flag as true all the time. What kind of files it will update if it needs to be a part of make update?

Can it be a separate script just used by whoever is working on cherry-pick?

**Comment by sonyafenge** [comment]
> I have a concern to add this as a generic component for make update. Cherry-pick is not regular process we are doing. Also CI tool cannot be updated to have this flag as true all the time. What kind of files it will update if it needs to be a part of make update?
> 
> Can it be a separate script just used by whoever is working on cherry-pick?

This will not impact CI and only when you add CHERRY_PICK=true in your local, related files will be updated. we don't need any changing for CI process. of cause, this is only one of the usage methods. we can definitely remove it from CI.

**Comment by Sindica** [comment]
> > I have a concern to add this as a generic component for make update. Cherry-pick is not regular process we are doing. Also CI tool cannot be updated to have this flag as true all the time. What kind of files it will update if it needs to be a part of make update?
> > Can it be a separate script just used by whoever is working on cherry-pick?
> 
> This will not impact CI and only when you add CHERRY_PICK=true in your local, related files will be updated. we don't need any changing for CI process. of cause, this is only one of the usage methods. we can definitely remove it from CI.

My point is: there is no reason to add this process into make update. Whoever is working on cherry-pick can simply run the script when necessary. Make update is an aggregated script that takes much longer, usually it does not need to be run at each cherry-pick. I am worrying about there could be some Arktos files (not from cherry-pick) that needs to be updated due to cherry-pick and be mistakenly added the note due to the switch is on.

File: hack/arktos_cherrypick.sh [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    git diff --name-only --diff-filter=A $MERGED_COMMIT $HEAD_COMMIT | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "perf-tests\/clusterloader2\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/metrics\/" | \
+        egrep -v "staging\/src\/k8s.io\/component-base\/version" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" > $LOGDIR/added_files
+    git diff --cached --name-only --diff-filter=A | \
+        egrep -v "\.git|\.md|\.bazelrc|\.json|\.pb|\.yaml|BUILD|boilerplate|vendor\/" | \
+        egrep -v "\.mod|\.sum|\.png|\.PNG|OWNERS|arktos_copyright" >> $LOGDIR/added_files || true
+    #grep -F -x -v -f $REPODIRNAME/hack/arktos_copyright_copied_k8s_files $LOGDIR/added_files_git > $LOGDIR/added_files_less_copied
```
Comment (vinaykul) [inline hack/arktos_cherrypick.sh:136]: we can remove these commented lines if they are not needed

File: hack/arktos_cherrypick.sh [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+if [ -z $1 ]
+then
+    setup_repos
+fi
+
+get_added_files_list
+get_modified_files_list
+
+add_arktos_cherrypick $LOGDIR/added_files $LOGDIR/changed_files
```
Comment (vinaykul) [inline hack/arktos_cherrypick.sh:208]: Does this end up adding cherrypick line to all added and modified files? Or does it apply only to the new files that were cherrypicked?

Comment (sonyafenge) [reply hack/arktos_cherrypick.sh:208]: This applied to both new added/updated files

**Comment by sonyafenge** [comment]
> > > I have a concern to add this as a generic component for make update. Cherry-pick is not regular process we are doing. Also CI tool cannot be updated to have this flag as true all the time. What kind of files it will update if it needs to be a part of make update?
> > > Can it be a separate script just used by whoever is working on cherry-pick?
> > 
> > 
> > This will not impact CI and only when you add CHERRY_PICK=true in your local, related files will be updated. we don't need any changing for CI process. of cause, this is only one of the usage methods. we can definitely remove it from CI.
> 
> My point is: there is no reason to add this process into make update. Whoever is working on cherry-pick can simply run the script when necessary. Make update is an aggregated script that takes much longer, usually it does not need to be run at each cherry-pick. I am worrying about there could be some Arktos files (not from cherry-pick) that needs to be updated due to cherry-pick and be mistakenly added the note due to the switch is on.

updated to directly run hack/arktos_cherry-pick.sh

File: hack/arktos_cherrypick.sh (hunk 1 above)
Comment (sonyafenge) [reply hack/arktos_cherrypick.sh]: removed these comments.

File: hack/arktos_cherrypick.sh (hunk 2 above)
Comment (vinaykul) [reply hack/arktos_cherrypick.sh:218]: As per yesterday's discussion, I understand this script will add cherry-pick comment only to newly added files, and it won't modify existing files that were added by us. If that is the case, it looks good to me.

**Pull request merged**